### DISCUSSION TITLE
Allow creating food items from shopping list

### DISCUSSION THREAD

**Discussion by koelle25** [mealie-recipes/mealie#3602] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When adding an item to a shopping list and enabling the "Food" option, it is only possible to choose from already existing foods. If the wanted item is not existing it has to be created first, which pulls you out of the shopping list again.

Creating foods is currently really cumbersome and needs the user to:
1. Click on the username at the top left
2. Scroll down, search for and click on "Manage data"
3. Click on "+ Create" to create a new food item

Afterwards:
1. Open the shopping lists page again
2. Select the shopping list
3. Click on "+ Add"
4. Enable the "Food" option again
5. Select the newly created food item

### Please provide a concise description of the feature that would resolve your issue.

When typing the wanted and not existing food item in the field, e.g.:
![grafik](https://github.com/mealie-recipes/mealie/assets/915514/ad12f3d1-b05d-4698-a6d6-0234ea420aa5)

It should be possible to instantly create this food item on the fly. One idea would be to have a little "+" button like in this interface I built for another application:
![grafik](https://github.com/mealie-recipes/mealie/assets/915514/7ff56a0f-9a4b-40dc-aa57-4cfc2396d8cb)

Clicking on this results in a popup like this:
![grafik](https://github.com/mealie-recipes/mealie/assets/915514/35d733ca-51af-406b-9fd5-9b7cdcc43bb1)

After creating the item it is added to the list and automatically selected:
![grafik](https://github.com/mealie-recipes/mealie/assets/915514/acae8d5f-6cf6-4d79-8bc0-3a131f847a4b)

### Please consider and list out some caveats or tradeoffs made in your design decision

The only thing coming to my mind right now is that users could unintentionally create duplicates. This could either be prevented or just being permitted, as the data management page allows merging food items in such cases.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by boc-the-git** [comment]
Hey @koelle25, I think this is perfectly reasonable.

It's not something I've looked into so I can't speak to whether there's any technical limitation for why it is as it is.

That said, if this is to be implemented it should be done in the same way as the creation works on the recipe editor page, as seen here.. (I guess what's not clear in the gif is me pressing enter to create the food, as the prompt advises)

![adding_food_via_enter](https://github.com/mealie-recipes/mealie/assets/3479092/d94a84b6-fdec-49f1-bc9a-335a05fde5c0)

**Reply by koelle25** [reply]
Ah yes, thank you! Didn't know that the feature itself is existing at this location. That's exactly what I meant.

Maybe it would be possible to "copy" this feature to the shopping list, too - that would be a somewhat easy implementation then.

**Reply by boc-the-git** [reply]
We'd definitely welcome (and appreciate!) a PR that does that :)

**Comment by michael-genson** [comment]
See: https://github.com/mealie-recipes/mealie/pull/3603

I've been meaning to address a different, though similar feature on the shopping list regarding labels, so I added both in this PR
